adapters/repos/db/lsmkv/bucket_recover_from_wal.go

```diff
@@ -75,18 +75,56 @@ func (b *Bucket) mayRecoverFromCommitLogs(ctx context.Context) error {
 		logOnceWhenRecoveringFromWAL.Do(func() {
 			b.logger.WithField("action", "lsm_recover_from_active_wal").
 				WithField("path", b.dir).
-				Warning("active write-ahead-log found. Did weaviate crash prior to this?")
+				Warning("active write-ahead-log found")
 		})
 	}
 
 	// recover from each log
-	for _, fname := range walFileNames {
+	for i, fname := range walFileNames {
 		path := filepath.Join(b.dir, strings.TrimSuffix(fname, ".wal"))
 
 		cl, err := newCommitLogger(path, b.strategy)
 		if err != nil {
 			return errors.Wrap(err, "init commit logger")
 		}
+
+		memtableExists, err := fileExists(path + ".mem")
+		if err != nil {
+			return err
+		}
+		if memtableExists {
+			walForActiveMemtable := i == len(walFileNames)-1
+
+			if walForActiveMemtable && !b.disableActiveMemtableReuse {
```

Comment:
Shouldn't this be independent from .wal recovery? I'd assume that we delete the .wal if we write the memtable to disk

Reply:
wal is maintained was it will continue to be used once the memtable is deserialized from disk

Reply:
In our previous attemp I had problems with writing to an existing .wal after closing and reopening